로그인

jaea-kim opened the pull request:
# 구현 내용
username 과 password로 로그인 
로그인 시 access token 생성
컨트롤러에서 @AutheticationPrincipal 을 통해 jwt 에 저장된 유저 정보를 접근할 수 있도록 함

## 남은 것 
refresh token

jaea-kim committed:
feat: 기본 로그인 구현 #6

jaea-kim committed:
feat: jjwt 의존성 추가 #6

jaea-kim committed:
feat: jwt access token 생성 및 filter 추가 #6

jaea-kim committed:
feat: @AuthenticationPrincipal에서 유저 정보를 받기 위해 설정 #6

- UserDetailService와 UserDetail 인터페이스를 확장해서 구현

jaea-kim committed:
feat: username에서 UserDetails 기반으로 개선 #6

derek-Flab commented on src/main/java/com/project/triplog/service/UserService.java:
```diff
@@ -44,4 +49,16 @@ private String encodingPassword(String password) {
+			throw new BadCredentialsException("비밀번호가 일치하지 않습니다.");
+		}
+
+		String token = jwtTokenProvider.createToken(user.getEmail());
```
메서드 구현체에는 username을 파라미터로 받는데 여기서는 email을 넘기고 있어서? 혹시 의도하신건가요? 
인증에서 username과 email의 용도에 대해서 한번 간단하게 설명해주실래요?

jaea-kim replied:
아니요 
이 부분은 제가 수정하다가 놓친 거 같습니다.
username( 현재는 userId) 만 파라미터로 받고 username 을 기준으로 인증하려고 했던 것이 맞습니다! 
해당 내용 수정했습니다!

jaea-kim committed:
Merge origin/feature/5-join and resolve conflicts #6

- Git 충돌 해결 완료

jaea-kim committed:
feat: userId 기반 인증으로 통일 #6

- UserController에 login API 추가
- 모든 인증 관련 코드를 userId 기반으로 완전 통일
- LoginRequest에서 userId 필드 사용
- User 도메인에서 username → userId 필드 변경
- UserRepository에서 findByUserId, existsByUserId 메서드 사용
- Security 관련 파라미터명까지 userId로 통일

jaea-kim committed:
Merge branch 'feature/5-join' into feature/6-login
